work/worker.go

```diff
@@ -750,25 +779,43 @@ CommitTransactionLoop:
 		// Start executing the transaction
 		env.state.SetTxContext(tx.Hash(), common.Hash{}, env.tcount)
 
-		err, logs := env.commitTransaction(tx, bc, rewardbase, vmConfig)
+		// Verify that tx is included in the bundle
+		targetBundle := &builder.Bundle{}
+		numShift := 1
+		if bundleIdx := builder_impl.FindBundleIdx(bundles, tx); bundleIdx != -1 {
+			targetBundle = bundles[bundleIdx]
+			numShift = len(targetBundle.BundleTxs)
+		}
+
+		if len(targetBundle.BundleTxs) != 0 {
+			atomic.StoreInt32(&isExecutingBundleTxs, 1)
+			err, tx, logs = env.commitBundleTransaction(targetBundle, bc, rewardbase, vmConfig)
```

Comment:
We must ensure that `commitBundleTransaction` always returns non-nil `tx` because it can be accessed.
I think always returning last executed tx (regardless of error or not) can be ok.

Reply:
Yes, I agree with you.
However, the first tx is a generator, and if gen fails there is no tx to return.
https://github.com/kaiachain/kaia/pull/272/files#diff-aa53dfe64c81ad7f4673af6857ca8ae7873bb18224202adb09f765ce4f458411R912

What about returning a Transaction pointer instead of nil? I mean `&types.Transaction{}`.

Reply:
Sounds great

Reply:
I've fixed it at f7474a6. Thank you!